apps/src/code-studio/components/progress/StageLockDialog.jsx

```diff
@@ -0,0 +1,284 @@
+import React from 'react';
+import Radium from 'radium';
+import { connect } from 'react-redux';
+import BaseDialog from '@cdo/apps/templates/BaseDialog';
```

Comment:
Generally speaking, it just takes an `onConfirm` callback that can do what
you want.  It also supports swapping in your own button components if you
prefer that (example from storybook here:
https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/templates/Dialog.jsx#L227
).

On Mon, Aug 8, 2016 at 11:52 AM, Brent Van Minnen <notifications@github.com>
wrote:

> In apps/src/code-studio/components/progress/StageLockDialog.jsx
> <https://github.com/code-dot-org/code-dot-org/pull/9760#discussion_r73932383>
> :
>
> > @@ -0,0 +1,284 @@
> > +import React from 'react';
> > +import Radium from 'radium';
> > +import { connect } from 'react-redux';
> > +import BaseDialog from '@cdo/apps/templates/BaseDialog';
>
> tagging @poorvasingal <https://github.com/poorvasingal> and @tanyaparker
> <https://github.com/tanyaparker> if they want to weigh in on the design
> decisions.
>
> Note - one other thing that's pretty different about my save button (in
> addition to the different color), is that it is an async operation and
> changes to "Saving..." while performing the operation. I guess we could
> make Dialog know how to handle this sort of thing too if we wanted.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/code-dot-org/code-dot-org/pull/9760/files/b74016256c91abfec19d89261415d5b56b0c8c7f#r73932383>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABinkeFu9OP-tdR9ThjaX92LqjLu2Sx4ks5qd3r5gaJpZM4JZxpO>
> .
>



-- 
Brad Buchanan
Software Engineer, Code.org
brad@code.org